client.go

```diff
@@ -291,6 +314,55 @@ func (c *Client) handleLine(line string) {
 			}
 		}
 	}
+	if strings.HasPrefix(line, ":") {
+		if strings.Contains(line, "tmi.twitch.tv JOIN") {
+			channel, username := parseJoinPart(line)
+
+			if username != c.ircUser && !isInSlice(username, c.channelUserlist[channel]) {
+				c.channelUserlist[channel] = append(c.channelUserlist[channel], username)
+			}
+
+			if c.onUserJoin != nil {
+				c.onUserJoin(channel, username)
+			}
+		}
+		if strings.Contains(line, "tmi.twitch.tv PART") {
+			channel, username := parseJoinPart(line)
+
+			c.channelUserlist[channel] = removeElement(username, c.channelUserlist[channel])
+
+			if c.onUserPart != nil {
+				c.onUserPart(channel, username)
+			}
+		}
+		if strings.Contains(line, "353 "+c.ircUser) {
+			channel, users := parseNames(line)
+
+			for _, user := range users {
+				c.channelUserlist[channel] = append(c.channelUserlist[channel], user)
+			}
+		}
+	}
+}
+
+func removeElement(elem string, slice []string) []string {
```

Comment:
This is very inefficent and should be just another map aswell. So a map inside a map in total. The value doesn't really matter in case of the second map.